src/Compilers/CSharp/Portable/BoundTree/UnboundLambda.cs

```diff
@@ -623,6 +628,28 @@ private BoundLambda ReallyInferReturnType(NamedTypeSymbol delegateType, Immutabl
             }
             lambdaSymbol.SetInferredReturnType(inferredReturnType.RefKind, returnType);
 
+            if (forErrorRecovery)
```

Comment:
>if (forErrorRecovery) [](start = 12, length = 21)

Is there a reason to do this work if inference failed?

Reply:
If there is still a chance that we have something unconverted within the bound lambda, then perhaps we want to bind for error recovery those parts without target type when the return type is ```LambdaSymbol.InferenceFailureReturnType```

---
In reply to: [328732473](https://github.com/dotnet/roslyn/pull/38750#discussion_r328732473) [](ancestors = 328732473)

Reply:
When binding during return type inference, there is nothing wrong with having some returned expressions unbound.  Binding them to any particular type would cause us to infer an incorrect result type.  Even in error recovery scenarios.

---
In reply to: [328733772](https://github.com/dotnet/roslyn/pull/38750#discussion_r328733772) [](ancestors = 328733772,328732473)